docs/ReleaseHistory.md

```diff
@@ -12,6 +12,12 @@
 - FPS => False positive reduction in static analysis.
 - FNS => False negative reduction in static analysis.
 
+# UNRELEASED
+- BRK, NEW: SecretMasker.MaskSecrets and ISecretMasker.DetectSecrets now accept an Action<Detection> callback (with a default value of null) to receive detections that result from the operation. This is binary-breaking only for SecretMasker callers and source-breaking as well for ISecretMasker implementations.
+- BRK: `Detection` is now sealed, and immutable. All property setters, and the copy constructor is removed.
+- BRK: `Detection` is no longer overrides Equals and GetHashCode nor implements `IEquatable<Detection>`. Equality comparison of `Detection` will now be based on reference equality.
```

Comment:
>Equals [](http://example.com/codeflow?start=42&length=6)

`Equals` and `GetHashCode`. this standard isn't a *foolish* consistency and so it isn't a hobgoblin of a little mind. nay, it is a prudent consistency. SEV3